lib/from_honeybee/geometry/aperture.rb

```diff
@@ -59,39 +59,80 @@ def to_openstudio(openstudio_model)
         os_vertices << OpenStudio::Point3d.new(vertex[0], vertex[1], vertex[2])
       end
       reordered_vertices = OpenStudio.reorderULC(os_vertices)
-
-      os_subsurface = OpenStudio::Model::SubSurface.new(reordered_vertices, openstudio_model)
-      os_subsurface.setName(@hash[:identifier])
-
-      # assign the construction if it exists
-      if @hash[:properties][:energy][:construction]
-        construction_identifier = @hash[:properties][:energy][:construction]
-        construction = openstudio_model.getConstructionByName(construction_identifier)
-        unless construction.empty?
-          os_construction = construction.get
-          os_subsurface.setConstruction(os_construction)
-        end
-      end
       
-      # assign the bondary condition object if it's a Surface
-      if @hash[:boundary_condition][:type] == 'Surface'
-        # get adjacent sub surface by identifier from openstudio model
-        adj_srf_identifier = @hash[:boundary_condition][:boundary_condition_objects][0]
-        sub_srf_ref = openstudio_model.getSubSurfaceByName(adj_srf_identifier)
-        unless sub_srf_ref.empty?
-          sub_srf = sub_srf_ref.get
-          os_subsurface.setAdjacentSubSurface(sub_srf)
+      # triangulate subsurface if neccesary
+      triangulated = false
+      final_vertices_list = []
+      if reordered_vertices.size > 4
+        
```

Comment:
Can we add in a check that, if the geometry has a `Surface` boundary condition and needs to be triangulated, we raise an exception for now? Since you are not currently mapping the multiple triangles to the new adjacent triangles, the resulting model for this case right now would not be valid/simulate-able. In Python, we have methods that get the adjacent surface and triangulate the neighbor simultaneously and don't have this issue. Until I can implement those methods here, I think it's safer to raise an exception.

Reply:
I could take a crack at implementing that, basically I would triangulate the first side, then copy reversed portions of the first window into the second zone (rather than re-triangulating the second sub surface).

Reply:
Probably want a unit test or two for this case.

Reply:
Can you make a JSON similar to `model_5vertex_sub_faces.json` but with the aperture and door on interior surfaces (e.g. that require surface matching)?